connectors/bitbucket-server/src/main/java/com/vmware/connectors/bitbucket/server/BitbucketServerController.java

```diff
@@ -0,0 +1,484 @@
+                                                  @RequestParam(COMMENT_PARAM_KEY) final String comment,
+                                                  @RequestHeader(AUTH_HEADER) final String authHeader,
+                                                  @RequestHeader(BASE_URL_HEADER) final String baseUrl) {
+
+        logger.info("Comment ACTION for bitbucket server pull request: {}, baseURL: {}",
+                pullRequest,
+                baseUrl);
+
+        final HttpHeaders headers = buildHeaders(authHeader);
+        headers.add(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE);
+
+        final URI uri = UriComponentsBuilder.fromHttpUrl(baseUrl + "/rest/api/1.0/projects/{PROJECT_KEY}/repos/{REPOSITORY_SLUG}/pull-requests/{PULL_REQUEST_ID}/comments")
+                .buildAndExpand(
+                        ImmutableMap.of(
+                                PROJECT_KEY, pullRequest.getProjectKey(),
+                                REPOSITORY_SLUG, pullRequest.getRepositorySlug(),
+                                PULL_REQUEST_ID, pullRequest.getPullRequestId()
+                        )
+                ).toUri();
+
+        final BitbucketServerComment bitBucketServerComment = new BitbucketServerComment(comment);
+        final ListenableFuture<ResponseEntity<String>> response = this.rest.exchange(
+                uri,
+                HttpMethod.POST,
+                new HttpEntity<>(bitBucketServerComment, headers),
+                String.class);
+        return Async.toSingle(response);
+    }
+
+    @GetMapping("/test-auth")
+    @SuppressWarnings("PMD.JUnit4TestShouldUseTestAnnotation") // PMD assumes the method as Junit Test if the API url has 'test'.
+    public Single<ResponseEntity<Void>> testAuth(@RequestHeader(AUTH_HEADER) final String authHeader,
+                                                 @RequestHeader(BASE_URL_HEADER) final String baseUrl) {
+        final HttpHeaders headers = new HttpHeaders();
+        headers.add(AUTHORIZATION, authHeader);
+
+        final ListenableFuture<ResponseEntity<JsonDocument>> response = this.rest.exchange(
+                baseUrl,
+                HttpMethod.GET,
+                new HttpEntity<>(headers),
+                JsonDocument.class);
+
+        return Async.toSingle(response)
+                .map(BitbucketServerController::stripBody)
+                .onErrorResumeNext(BitbucketServerController::map404To200);
+    }
+
+    private static ResponseEntity<Void> stripBody(final ResponseEntity<JsonDocument> entity) {
+        return ResponseEntity.status(entity.getStatusCode()).build();
+    }
+
+    private static Single<ResponseEntity<Void>> map404To200(final Throwable throwable) {
+        if (throwable instanceof HttpClientErrorException &&
+                HttpClientErrorException.class.cast(throwable).getStatusCode() == HttpStatus.NOT_FOUND) {
+            // If Bitbucket server is down, then we return 200 back to the client.
```

Comment:
If the comment is accurate (I'm not sure it is), I wouldn't do this.  Let the client decide what to do (block the user from saving credentials or save it but realize they might want to check again later).

Reply:
Thanks for pointing it out.. Changed the comment to

```
If Bitbucket server base url is incorrect, then we return 200 back to the client.
```

If the base url is not correct, then it will throw 404. In this case, we are mapping it 200.